keps/sig-api-machinery/1027-api-unions/README.md

```diff
@@ -133,27 +258,30 @@ type TopLevelUnion struct {
 }
 
 // This will generate one union, with two fields and a discriminator.
-// +union
 type Union struct {
 	// +unionDiscriminator
-	// +optional
+        // +unionAllowEmpty
+        // +required
 	UnionType string `json:"unionType"`
 
+        // +unionMember
 	// +optional
 	FieldA int `json:"fieldA"`
-    // +optional
+        // +unionMember
+        // +optional
 	FieldB int `json:"fieldB"`
 }
 
-// This also generates one union, with two fields and on discriminator.
+// This generates two unions, each with two fields and a discriminator. The 
 type Union2 struct {
 	// +unionDiscriminator
+        // +required
 	Type string `json:"type"`
-	// +unionDeprecated
-    // +optional
+	// +unionMember=type
```

Comment:
...I was expecting this to show the value which you must set the discriminator to in order to choose this leg of the union, not to say which discriminator this field belongs with.

Reply:
clarified above